the-collab-lab/website-og
# fix: restore Stripe donation buttons

## Summary
This PR reintroduces the Stripe donation buttons, and does a few optimization chores in the process.

**Fixes:**
- Restores the Stripe one-time donation buttons when in a Netlify environment

**Chores:**
- takes advantage of Netlify context-dependent variables 
- moves network requests out of `pages.11tydata.js` and into their own global data files
  - renames `pages.11tydata.js` to `pages.11tydata.json` because its only export is static
- renames the `getDonatonOptions` data object to `donations` for clarity – it's a computed object, not a getter
- prefetches the DNS resolution for the Stripe dot com domain
- Updates the styles of the Stripe donation buttons to better match the github donation button

### Why use unique `_data/` files?
Template data files like`pages.11tyData.js`, and any network requests within them, are executed as Eleventy builds for _each_ template in the their corresponding directory. We only need to get the Stripe and GraphQL data once, so these requests are better suited to global data files.

## Test plan
1. Visit the [deploy preview](https://deploy-preview-243--the-collab-lab.netlify.app/)
2. Verify that the Stripe donation buttons appear in the footer
3. Click a donation button
4. Verify that you are redirected to a **test** checkout page corresponding to that donation amount
